tests/test_geoserver.py

```diff
@@ -46,17 +63,90 @@ def get_geoserver_settings():
+                                                   datastore_path=geoserver_workspace_path)
+
+    # Preparations needed to make tests work without all the other handlers running
+    datastore_path = get_datastore_path(test_instance.workspace_folders[workspace_key])
+
+    # This next part can fail if the user running this test doesn't have write access to the directory
+    copy_shapefile(basename=test_instance.test_shapefile_name, destination=datastore_path)
+
+    return workspace_name, datastore_name
+
+
+def reset_geoserver_test_workspace(test_instance, geoserver_handler):
+    """
+    Removes a workspace on the file system and the associated resources on Geoserver.
+    """
+    for _, workspace in test_instance.workspaces.items():
+        try:
+            geoserver_handler._remove_workspace_request(workspace_name=workspace)
+        except GeoserverError:
+            # Making sure all test workspaces are removed
+            pass
+    for _, folder in test_instance.workspace_folders.items():
+        try:
+            if not folder.startswith("/tmp/"):
+                raise PermissionError("Aborting test workspace reset. The test workspace path should be in the `/tmp`"
+                                      f"directory, but was instead found at the path `{folder}`.")
+            # Make sure access permissions are enabled before deleting files
+            os.chmod(folder, 0o777)
+            for root, dirs, files in os.walk(folder):
+                for resource in dirs + files:
+                    os.chmod(os.path.join(root, resource), 0o777)
+            shutil.rmtree(folder)
+        except FileNotFoundError:
+            pass
+
+
 def copy_shapefile(basename, destination):
     full_filename = f"{COWBIRD_ROOT}/tests/resources/{basename}"
     Path(destination).mkdir(parents=True, exist_ok=False)
     for file in glob.glob(f"{full_filename}.*"):
         shutil.copy(file, destination)
 
 
-@pytest.mark.online
-@pytest.mark.geoserver
-class TestGeoserverRequests:
+def get_datastore_path(workspace_path):
+    return workspace_path + "/shapefile_datastore"
+
+
+class TestGeoserver:
     geoserver_settings = get_geoserver_settings()
+    workspaces = {}
+    workspace_folders = {}
+
+    test_shapefile_name = "Espace_Vert"
+
+    def teardown_class(self):
+        # Couldn't pass fixture to teardown function.
+        teardown_gs = Geoserver(settings={}, name="Geoserver", **self.geoserver_settings)
+        teardown_gs.ssl_verify = self.geoserver_settings["ssl_verify"]
+        reset_geoserver_test_workspace(self, teardown_gs)
+        utils.clear_handlers_instances()
+
+    @staticmethod
+    def get_geoserver():
+        geoserver = Geoserver(settings={}, name="Geoserver", **TestGeoserver.geoserver_settings)
+        geoserver.ssl_verify = TestGeoserver.geoserver_settings["ssl_verify"]
+        return geoserver
+
+
+@pytest.mark.geoserver
+class TestGeoserverRequests(TestGeoserver):
```

Comment:
Missing online marker? I'm assuming "Requests" here uses a running GeoServer (or mock). If not, disregard this comment.

Reply:
You are right, both TestGeoserver classes would be "online" since they require a running instances of Geoserver and/or Magpie.